### PR TITLE
Set thread priority and stack size via configuration

### PR DESCRIPTION
**Pull request by pyhys**

### CODE REVIEW

**Review by olbjo — COMMENTED** [review]
👍

**Review by olbjo — APPROVED** [review]

File: src/ports/linux/pnal_options.h.in [hunk 1]
```diff
@@ -18,14 +18,4 @@
 
 #include <net-snmp/net-snmp-config.h>
 
-/** Priority of SNMP thread (if SCHED_FIFO is used) */
```
Comment (hefloryd) [inline src/ports/linux/pnal_options.h.in:21]: This file is now empty so let's remove it and update cmakefile.

File: src/ports/rt-kernel/sampleapp_main.c [hunk 2]
```diff
@@ -194,6 +194,7 @@ int main (void)
    pnet_default_cfg.cb_arg = (void *)&appdata;
    strcpy (pnet_default_cfg.station_name, appdata.arguments.station_name);
    strcpy (pnet_default_cfg.file_directory, APP_DEFAULT_FILE_DIRECTORY);
+   /* Note: threads_cfg not is used for rt-kernel (initialized to 0) */
```
Comment (hefloryd) [inline src/ports/rt-kernel/sampleapp_main.c:197]: This should be fixed.

**Review by olbjo — APPROVED** [review]

File: src/ports/linux/pnal_options.h.in (hunk 1 above)
Comment (pyhys) [reply src/ports/linux/pnal_options.h.in:21]: Done

File: src/ports/rt-kernel/sampleapp_main.c (hunk 2 above)
Comment (pyhys) [reply src/ports/rt-kernel/sampleapp_main.c:197]: Assert in rt-kernel code to make sure it not is set.

File: src/ports/linux/pnal_config.h [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include <stddef.h>
+
+/**
+ * Thread priority and stack size
+ */
+typedef struct pnet_thread_cfg
```
Comment (hefloryd) [inline src/ports/linux/pnal_config.h:37]:
```suggestion
typedef struct pnal_thread_cfg
```

Comment (pyhys) [reply src/ports/linux/pnal_config.h]: Done

File: src/ports/rt-kernel/sampleapp_main.c (hunk 2 above)
Comment (pyhys) [reply src/ports/rt-kernel/sampleapp_main.c]: Implemented operating-system dependent pnal configuration.

**Pull request merged**